redis/asyncio/client.py

```diff
@@ -255,7 +255,7 @@ def __init__(
 
         self.response_callbacks = CaseInsensitiveDict(self.__class__.RESPONSE_CALLBACKS)
 
-        if self.connection_pool.connection_kwargs.get("protocol") == "3":
+        if self.connection_pool.connection_kwargs.get("protocol") in ["3", 3]:
```

Comment:
I wonder if it's cheaper to do:
```        if str(self.connection_pool.connection_kwargs.get("protocol")) == '3'```

I think it would be, because this way there's no O(n) on the check. WDYT>

Reply:
I checked it before, my approach is faster

Reply:
Agreed. Surprised.

For the PR curious at scale of 1M attempts these are the numbers:

conv : 1.6109482662217037e-07
conv : 4.0583412699501565e-08

The latter is @dvora-h implementation.